app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddRoadSurface.kt

```diff
@@ -3,27 +3,30 @@ package de.westnordost.streetcomplete.quests.surface
 import de.westnordost.streetcomplete.R
 import de.westnordost.streetcomplete.data.meta.ANYTHING_UNPAVED
 import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
-import de.westnordost.streetcomplete.data.osm.osmquest.SimpleOverpassQuestType
 import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
 import de.westnordost.streetcomplete.data.osm.mapdata.OverpassMapDataAndGeometryApi
+import de.westnordost.streetcomplete.data.osm.osmquest.SimpleOverpassQuestType
 import de.westnordost.streetcomplete.settings.ResurveyIntervalsStore
 
 
-class AddRoadSurface(o: OverpassMapDataAndGeometryApi, r: ResurveyIntervalsStore)
-    : SimpleOverpassQuestType<String>(o) {
-
+class AddRoadSurface(o: OverpassMapDataAndGeometryApi, r: ResurveyIntervalsStore) : SimpleOverpassQuestType<DetailSurfaceAnswer>(o) {
     override val tagFilters = """
         ways with highway ~ ${ROADS_WITH_SURFACES.joinToString("|")}
-         and (
-           !surface
-           or surface ~ ${ANYTHING_UNPAVED.joinToString("|")} and surface older today -${r * 4} years
-           or surface older today -${r * 12} years
-         )
-         and (access !~ private|no or (foot and foot !~ private|no))
+        and
+        (
+            !surface
+            or surface ~ ${ANYTHING_UNPAVED.joinToString("|")} and surface older today -${r * 4} years
+            or surface older today -${r * 12} years
+        or
+            (
+            surface ~ paved|unpaved
+            and !surface:note
+            and segregated != yes
+            )
+        )
+        and (access !~ private|no or (foot and foot !~ private|no))
```

Comment:
indentation is a bit off. I first thought that the `or` would conflict with the `and`